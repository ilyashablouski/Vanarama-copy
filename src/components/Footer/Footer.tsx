import React, { FC } from 'react';
import dynamic from 'next/dynamic';
import { LazyLoadComponent } from 'react-lazy-load-image-component';
import ReactMarkdown from 'react-markdown';
import { PrimaryFooter_primaryFooter as PrimaryFooter } from '../../../generated/PrimaryFooter';
import RouterLink from '../RouterLink/RouterLink';

const Text = dynamic(() => import('core/atoms/text'));
const FooterColumn = dynamic(() => import('./FooterColumn'));

interface IFooter {
  primaryFooter: PrimaryFooter;
}

const Footer: FC<IFooter> = ({ primaryFooter }) => {
  const { linkGroups, legalStatement } = primaryFooter;

  return (
    <LazyLoadComponent
      visibleByDefault={typeof window === 'undefined'}
      threshold={250}
    >
      <footer className="footer">
        {linkGroups?.map(linkGroup => (
          <FooterColumn key={linkGroup?.name || ''} linkGroup={linkGroup} />
        ))}
        <hr className="footer--divider -fullwidth" />
        {legalStatement?.title && (
          <Text color="white" size="xsmall" tag="p">
            {legalStatement?.title}
          </Text>
        )}
        {legalStatement?.body && (
          <div className="-text-columns">
            <Text size="xsmall" color="medium">
              <ReactMarkdown
                source={legalStatement.body}
                renderers={{
                  link: props => {
                    const { href, children } = props;
                    return <RouterLink link={{ href, label: children }} />;
                  },
                }}
              />
            </Text>
          </div>
        )}
      </footer>
    </LazyLoadComponent>
  );
};

export default React.memo(Footer);
