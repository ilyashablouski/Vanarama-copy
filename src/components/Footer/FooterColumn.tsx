import React, { FC } from 'react';
import ReactMarkdown from 'react-markdown';
import Heading from '@vanarama/uibook/lib/components/atoms/heading';
import Text from '@vanarama/uibook/lib/components/atoms/text';
import { PrimaryFooter_primaryFooter_linkGroups as LinkGroups } from '../../../generated/PrimaryFooter';
import RouterLink from '../RouterLink/RouterLink';

interface IFooterColumn {
  linkGroup: LinkGroups | null;
}

const FooterColumn: FC<IFooterColumn> = props => {
  const { linkGroup } = props;

  if (linkGroup) {
    return (
      <div className="footer--column" key={linkGroup.name || ''}>
        <Heading color="medium" tag="span" size="small">
          {linkGroup.name}
        </Heading>
        {!(linkGroup.links?.length || linkGroup.linkGroups?.length) &&
          linkGroup.body && <ReactMarkdown source={linkGroup.body} />}
        {linkGroup.links && (
          <ul>
            {linkGroup.links?.map(link => (
              <li key={link?.text || ''}>
                <RouterLink
                  link={{ href: link?.url || '', label: link?.text || '' }}
                  classNames={{ color: 'white', size: 'small' }}
                />
              </li>
            ))}
          </ul>
        )}
        {linkGroup.linkGroups && (
          <ul>
            {linkGroup.linkGroups?.map(el => (
              <li className={el?.links ? '-mb-300' : ''} key={el?.name || ''}>
                <Text size="small" color="medium" tag="span">
                  {el?.name}
                </Text>{' '}
                {el?.links?.map(link =>
                  link?.url ? (
                    <a
                      className="link -white -small"
                      href={link?.url || ''}
                      key={link?.url || ''}
                    >
                      {link?.text}
                    </a>
                  ) : (
                    <Text
                      size="small"
                      color="white"
                      tag="span"
                      key={link?.text || ''}
                    >
                      {link?.text}
                    </Text>
                  ),
                )}
              </li>
            ))}
          </ul>
        )}
      </div>
    );
  }
  return <></>;
};

export default FooterColumn;
