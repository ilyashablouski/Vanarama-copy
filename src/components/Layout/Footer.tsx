
export default () => (
  <footer className="Footer">
    &copy; Vanarama
  </footer>
)
