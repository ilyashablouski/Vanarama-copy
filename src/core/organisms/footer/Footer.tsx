import React, { FC, memo } from 'react';
import cx from 'classnames';

import { ILink } from 'core/interfaces/link';
import Heading from '../../atoms/heading';
import Link from '../../atoms/link';
import Text from '../../atoms/text';
import Logo from '../../atoms/logo';

import { IFooterProps } from './interfaces';
import Button from '../../atoms/button';

import LogoFacebook from '../../assets/icons/LogoFacebook';
import LogoTwitter from '../../assets/icons/LogoTwitter';
import LogoInstagram from '../../assets/icons/LogoInstagram';
import LogoLinkedin from '../../assets/icons/LogoLinkedin';
import LogoYoutube from '../../assets/icons/LogoYoutube';

const Footer: FC<IFooterProps> = memo(props => {
  const {
    className,
    emailAddress,
    phoneNumber,
    openingTimes,
    headingZero,
    headingOne,
    headingTwo,
    headingThree,
    linkGroupOne,
    linkGroupTwo,
    linkGroupThree,
    legalText,
  } = props;

  const actualYear = new Date().getFullYear();

  const renderLinkGroup = (
    linkGroup: ILink[],
    heading: string | null = null,
  ) => (
    <div className="footer--column">
      {heading && (
        <Heading size="small" color="medium">
          {heading}
        </Heading>
      )}
      <ul>
        {linkGroup.map(entry => (
          <li key={entry.label}>
            <Link color="white" size="regular" plain href={entry.href}>
              {entry.label}
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );

  const renderSocialLink = () => (
    <div className="footer--social-icons">
      <Button
        color="medium"
        size="small"
        fill="outline"
        round
        icon={<LogoFacebook />}
        iconPosition="before"
      />
      <Button
        color="medium"
        size="small"
        fill="outline"
        round
        icon={<LogoTwitter />}
        iconPosition="before"
      />
      <Button
        color="medium"
        size="small"
        fill="outline"
        round
        icon={<LogoInstagram />}
        iconPosition="before"
      />
      <Button
        color="medium"
        size="small"
        fill="outline"
        round
        icon={<LogoLinkedin />}
        iconPosition="before"
      />
      <Button
        color="medium"
        size="small"
        fill="outline"
        className="-round"
        icon={<LogoYoutube />}
        iconPosition="before"
      />
    </div>
  );

  return (
    <footer className={cx('footer', className)}>
      <div className="footer--column">
        {headingZero && (
          <Heading size="small" color="medium">
            {headingZero}
          </Heading>
        )}
        <ul>
          {phoneNumber && (
            <li className="-mb-300">
              <Text size="small" color="medium">
                Phone
              </Text>{' '}
              <Link color="white" href={`tel:${phoneNumber}`} size="small">
                {phoneNumber}
              </Link>
            </li>
          )}
          {openingTimes &&
            openingTimes.map(entry => (
              <li key={entry.day}>
                <Text size="small" color="medium" key={`${entry.day}_day`}>
                  {entry.day}
                </Text>{' '}
                <Text size="small" color="white" key={`${entry.time}_time`}>
                  {entry.time}
                </Text>
              </li>
            ))}
          <br />
          {emailAddress && (
            <li className="-mt-300">
              <Link
                plain
                size="small"
                color="white"
                href={`mail:${emailAddress}`}
              >
                {emailAddress}
              </Link>
            </li>
          )}
        </ul>
        <li>{renderSocialLink()}</li>
        <Logo asset="bvrla" />
      </div>

      {/* Link Group One */}
      {linkGroupOne && renderLinkGroup(linkGroupOne, headingOne)}

      {/* Link Group Two */}
      {linkGroupTwo && renderLinkGroup(linkGroupTwo, headingTwo)}

      {/* Link Group Three */}
      {linkGroupThree && renderLinkGroup(linkGroupThree, headingThree)}
      <div className="footer--divider -fullwidth" />
      <Text tag="p" size="xsmall" color="light" className="footer--copyright">
        &copy;
        {actualYear} Autorama UK LTD. All rights reserved.
      </Text>
      {legalText && (
        <Text tag="p" size="xsmall" color="medium">
          {legalText}
        </Text>
      )}
    </footer>
  );
});

export default Footer;
