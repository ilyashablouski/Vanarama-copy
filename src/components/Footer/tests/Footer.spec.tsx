import React from 'react';
import preloadAll from 'jest-next-dynamic';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

describe('<Footer />', () => {
  beforeEach(async () => {
    await preloadAll();
  });
  it('renders correctly with body', () => {
    const getComponent = () => {
      return renderer
        .create(
          <Footer
            primaryFooter={{
              id: '2zfD36955SlVBcnHfpsxnK',
              linkGroups: [
                {
                  name: 'GET IN TOUCH WITH US',
                  body:
                    '- __Phone__ 01442 838 195\n- __Monday–Thursday__ 8:30am–7pm\n- __Friday__ 9am–5pm\n- __Saturday__ 9am–3pm\n- __Sunday__ Closed\n- [enquiries@vanarama.co.uk](mailto:enquiries@vanarama.co.uk "enquiries@vanarama.co.uk")\n\n- Vanarama on Facebook\n- Vanarama on Twitter\n- Vanarama on Instagram\n- Vanarama on LinkedIn\n- Vanarama on Youtube\n\n- BVRLA SVG Logo',
                  links: null,
                  linkGroups: [
                    {
                      name: 'Phone',
                      links: [
                        {
                          text: '01442 838 195',
                          url: '01442 838 195',
                        },
                      ],
                    },
                    {
                      name: 'Monday–Thursday',
                      links: [
                        {
                          text: '8:30am–7pm',
                          url: null,
                        },
                      ],
                    },
                    {
                      name: 'Friday',
                      links: [
                        {
                          text: '9am–5pm',
                          url: null,
                        },
                      ],
                    },
                    {
                      name: 'Saturday ',
                      links: [
                        {
                          text: '9am–3pm',
                          url: null,
                        },
                      ],
                    },
                    {
                      name: 'Sunday',
                      links: [
                        {
                          text: 'Closed',
                          url: null,
                        },
                      ],
                    },
                    {
                      name: null,
                      links: [
                        {
                          text: 'enquiries@vanarama.co.uk',
                          url: 'enquiries@vanarama.co.uk',
                        },
                      ],
                    },
                  ],
                },
                {
                  name: 'VANS & TRUCKS',
                  body:
                    '- Van Leasing\n- Pickup Truck Leasing\n- Van Reviews\n- Van News\n- Compare Vans\n- Van Leasing Explained\n- Ask Us A Question\n- Advanced Breakdown Cover\n- Service Plus\n- Van Insurance\n- Van Insurance FAQs',
                  links: [
                    {
                      url: 'https://beta.vanarama.com/van-leasing.html',
                      text: 'Van Leasing',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/pickup-truck-leasing.html',
                      text: 'Pickup Truck Leasing',
                    },
                    {
                      url: 'https://beta.vanarama.com/van-reviews.html',
                      text: 'Van Reviews',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/latest-news/category/van-news.html',
                      text: 'Van News',
                    },
                    {
                      url: 'https://www.vanarama.com/compare-vans.html',
                      text: 'Compare Vans',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/van-leasing-explained.html',
                      text: 'Van Leasing Explained',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/van-leasing-questions.html',
                      text: 'Customer Leasing Questions',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/advanced-breakdown-cover.html',
                      text: 'Advanced Breakdown Cover',
                    },
                    {
                      url: 'https://beta.vanarama.com/serviceplus.html',
                      text: 'Service Plus',
                    },
                    {
                      url: 'https://beta.vanarama.com/van-insurance.html',
                      text: 'Van Insurance',
                    },
                    {
                      url: 'https://beta.vanarama.com/van-insurance/faq.html',
                      text: 'Van Insurance FAQs',
                    },
                  ],
                  linkGroups: null,
                },
                {
                  name: 'CARS',
                  body: null,
                  links: [
                    {
                      url: 'https://beta.vanarama.com/car-leasing.html',
                      text: 'Car Leasing',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/car-leasing/blog/category/car-news.html',
                      text: 'Car News',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/car-leasing-explained.html',
                      text: 'Car Leasing Explained ',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/redundancy-and-life-event-cover.html',
                      text: 'Redundancy & Life Event Cover',
                    },
                    {
                      url: 'https://beta.vanarama.com/eligibility-checker.html',
                      text: 'Eligability Checker',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/car-leasing-explained/given-company-cash-allowance-vs-company-car.html',
                      text: 'Company Cash Allowance',
                    },
                  ],
                  linkGroups: null,
                },
                {
                  name: 'ABOUT US & CONTACT',
                  body:
                    '- Our Price Promise\n- 5 Star Service Promise\n- Customer Leasing Reviews\n- Award Winning Leasing\n- Careers\n- Our Local Representatives\n- Community\n- FCA Initial Disclosure Document\n- Complaints Policy\n- Cookie & Privacy Policy\n- Terms & Conditions\n- Modern Slavery Statement\n- Sitemap',
                  links: [
                    {
                      url:
                        'https://beta.vanarama.com/about-us/price-protection-guarantee.html',
                      text: 'Our Price Promise',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/about-us/five-star-service.html',
                      text: '5 Star Service Promise',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/about-us/customer-testimonials.html',
                      text: 'Customer Leasing Reviews',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/latest-news/the-sunday-times-100-best-small-companies-to-work-for.html',
                      text: 'Award Winning Leasing',
                    },
                    {
                      url: 'https://beta.vanarama.com/careers.html',
                      text: 'Careers',
                    },
                    {
                      url: 'https://beta.vanarama.com/locations.html',
                      text: 'Our Local Representatives',
                    },
                    {
                      url: 'https://beta.vanarama.com/community.html',
                      text: 'Community',
                    },
                    {
                      url: 'https://beta.vanarama.com/fca.html',
                      text: 'FCA Initial Disclosure Document',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/contact-us/complaints-procedure.html',
                      text: 'Complaints Policy',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/legal/privacy-policy.html',
                      text: 'Cookie & Privacy Policy',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/legal/terms-and-conditions',
                      text: 'Terms & Conditions',
                    },
                    {
                      url:
                        'https://beta.vanarama.com/legal/modern-slavery-statement.html',
                      text: 'Modern Slavery Statement',
                    },
                    {
                      url: 'https://beta.vanarama.com/sitemap.html',
                      text: 'Sitemap',
                    },
                  ],
                  linkGroups: null,
                },
              ],
              legalStatement: {
                name: 'Global Disclaimer',
                title: '©2020 Autorama UK LTD. All rights reserved.',
                body:
                  'Autorama UK Ltd Registered in England and Wales with registered number: 05137709 Registered office: Vanarama, Maylands Avenue, Hemel Hempstead, HP2 7DE, United Kingdom. Vanarama, Vanarama Cars, Pickup Trucks Direct, Vanarama Short Term Leasing and Vanarama Insurance Services are trading styles of Autorama UK Ltd, who are authorised and regulated by the Financial Conduct Authority (Our Financial Conduct Authority Register number is 630748). VAT number GB842814720. Autorama UK Ltd is a Credit Broker not a Lender, we can introduce you to a selected group of Lenders. Calls are recorded for compliance and training purposes.',
              },
            }}
          />,
        )
        .toJSON();
    };

    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });
});
