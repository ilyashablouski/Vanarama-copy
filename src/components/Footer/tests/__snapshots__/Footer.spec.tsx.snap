// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly with body 1`] = `
<footer
  className="footer"
>
  <span
    className=""
    style={
      Object {
        "display": "inline-block",
      }
    }
  />
</footer>
`;
